### ISSUE TITLE
Error: Class incorrectly implements interface with no reason specified

### ISSUE THREAD

**Issue by hessryanm** [Microsoft/TypeScript#5751]
Creating new issue because the other one is closed/fixed.

Relates to issue #1734.  I just ran into this problem again today.

See Gist https://gist.github.com/hessryanm/177b0cada8c51516371e

Error is models/Card.ts(13,14): error TS2420: Class 'Card' incorrectly implements interface 'ICard'.

**Comment by RyanCavanaugh** [comment]
See also #5735. Can you reduce this to a self-contained repro (i.e. one that doesn't require a bunch of references to other .d.ts files)?

**Comment by hessryanm** [comment]
Yep, after I posted this I figured out what the problem was - it's fairly simple.

For the record, after making this issue I realized I had read the handbook about classes/interfaces wrong, so I realize that the code I am about to post is wrong, but the issue in terms of it not explaining the problem stands.

```
interface Interface {
  new (): Interface;
}

class Class implements Interface {
  constructor () {}
}
```

Basically, the problem with my code was that it's the *class* that has the `new` constructor, but an *instance* that is being checked against the interface.
